src/input/mod.rs

```diff
@@ -1552,6 +1561,31 @@ impl State {
         let button_state = event.state();
 
         if ButtonState::Pressed == button_state {
+            let mods = self.niri.seat.get_keyboard().unwrap().modifier_state();
+            let modifiers = modifiers_from_state(mods);
+
+            if self.niri.mods_with_mouse_binds.contains(&modifiers) {
+                let comp_mod = self.backend.mod_key();
+
+                if let Some(bind) = match button {
+                    BTN_LEFT => Some(Trigger::MouseLeft),
+                    BTN_RIGHT => Some(Trigger::MouseRight),
+                    BTN_MIDDLE => Some(Trigger::MouseMiddle),
+                    // Chromium treats these as equivalent: https://chromium.googlesource.com/chromium/src/+/refs/heads/main/ui/ozone/platform/wayland/host/wayland_pointer.cc
+                    BTN_BACK | BTN_SIDE => Some(Trigger::MouseBack),
+                    BTN_FORWARD | BTN_EXTRA => Some(Trigger::MouseForward),
```

Comment:
This actually doesn't work for my own mouse (when running in a tty, running through winit works because of the aforementioned support for both values). My mouse is a logitech G102 (`046d:c092 Logitech, Inc. G102/G203 LIGHTSYNC Gaming Mouse`), worth mentioning I'm using [libratbag](https://github.com/libratbag/libratbag), I don't think it affects it but there's no way for me to check

Reply:
I opened https://github.com/Smithay/smithay/pull/1611 to add these, I think it'll useful to support arbitrary keycodes, some mice have more than 2 additional buttons, and for instance mine has a DPI button that shows up as `BTN_FORWARD`, so they should take priority over the known ones. Unfortunately they are not syms like keys and I afaik there's no way to go from a name to a code other than to hard code all of them, maybe something like `Mouse115` (hex) or `Mouse277` (decimal) could work?

Reply:
Hmm, how do other WMs spell it? Also I guess this can be a follow-up PR.

Reply:
Turns out it's possible to convert them from names to codes, [river does so with libevdev](https://codeberg.org/river/river/src/commit/5ca829bd5acea4c68abcf16332ccfe6202da4aeb/river/command/map.zig#L244-L252). 

Hyprland uses `mouse:277` ([docs](https://wiki.hyprland.org/Configuring/Binds/#mouse-buttons)).

I agree this can be a follow-up PR.